src/Client.php

```diff
@@ -276,11 +276,11 @@ public function refundOrder(string $order_id)
             throw new InvalidOrderStatusException($order->getStatus()->get(), 'completed');
         }
 
-        if (!$order->getCurrentTransaction()->isCaptured()) {
+        if ($order->getCurrentTransaction()->isCapturable() && !$order->getCurrentTransaction()->isCaptured()) {
             throw  new RefundFailedException('Order is not yet captured, only captured order could be refunded');
         }
 
-        return $this->api_client->refundOrder(id: $order_id, orderData: ['order_lines' => $order->getOrderLines()->toArray()]);
+        return $this->api_client->refundOrder(id: $order_id, orderData: ['amount' => $order->getAmount()->get(), "description" => "Order refund", 'order_lines' => $order->getOrderLines()->toArray()]);
```

Comment:
Refund for Klarna pay later return error: `The browser (or proxy) sent a request that this server could not understand.`
So, such extending of orderData help to solve this error